crates/apollo-router-core/src/query_planner/model.rs

```diff
@@ -386,43 +386,46 @@ async fn fetch_node<'a>(
             Some(Response {
                 data, mut errors, ..
             }) => {
-                if let Some(entities) = data.get("_entities") {
-                    tracing::trace!(
-                        "Received entities: {}",
-                        serde_json::to_string(entities).unwrap(),
-                    );
-                    if let Some(array) = entities.as_array() {
-                        let mut response = response
-                            .lock()
-                            .instrument(tracing::trace_span!("response_lock_wait"))
-                            .await;
-
-                        let span = tracing::trace_span!("response_insert");
-                        let _guard = span.enter();
-                        for (i, entity) in array.iter().enumerate() {
-                            response.insert_data(
-                                &current_dir.join(Path::from(i.to_string())),
-                                entity,
-                            )?;
+                if let Value::Object(mut map) = data {
+                    if let Some(entities) = map.remove("_entities") {
+                        tracing::trace!(
+                            "Received entities: {}",
+                            serde_json::to_string(&entities).unwrap(),
+                        );
+
+                        if let Value::Array(mut array) = entities {
+                            let mut response = response
+                                .lock()
+                                .instrument(tracing::trace_span!("response_lock_wait"))
+                                .await;
+
+                            let span = tracing::trace_span!("response_insert");
+                            let _guard = span.enter();
+                            for (i, entity) in array.drain(..).enumerate() {
```

Comment:
not directly related to your pr, but array can be consumed here

```suggestion
                        if let Value::Array(array) = entities {
                            let mut response = response
                                .lock()
                                .instrument(tracing::trace_span!("response_lock_wait"))
                                .await;

                            let span = tracing::trace_span!("response_insert");
                            let _guard = span.enter();
                            for (i, entity) in array.into_iter().enumerate() {
```

Reply:
fixed